service/Service.AspNetCore/WebAPIEndpoints.cs

```diff
@@ -296,6 +299,78 @@ await service.DeleteDocumentAsync(documentId: documentId, index: index, cancella
         if (authFilter != null) { route.AddEndpointFilter(authFilter); }
     }
 
+    public static void AddPostDownloadEndpoint(this IEndpointRouteBuilder builder, string apiPrefix = "/", IEndpointFilter? authFilter = null)
+    {
+        RouteGroupBuilder group = builder.MapGroup(apiPrefix);
+
+        // File download endpoint
+        var route = group.MapPost(Constants.HttpDownloadEndpoint, async Task<IResult> (
```

Comment:
I would make some changes to this endpoint:
* use GET method, passing all params as query. GET is much easier to test, is cacheable, and is usually how files are downloaded
* I would remove the custom headers DocID, Volume etc. Why are they needed?
* There should be Content Type and Content Length headers. The type can be fetched from memory or using MimeTypesDetection class.
* There's some confusion between "document" and "file". The new feature allows to download one file, so I would avoid using the "document" terminology, e.g. `ExportDocumentAsync` should be `ExportFileAsync`
* About status codes, why turn `KernelMemoryException` into a 400 rather than a 503?